internal/services/v1alpha1/errors.go

```diff
@@ -0,0 +1,52 @@
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/authzed/spicedb/internal/namespace"
+	"github.com/authzed/spicedb/internal/services/shared"
+	"github.com/authzed/spicedb/internal/sharederrors"
+	"github.com/authzed/spicedb/pkg/datastore"
+	cerror "github.com/authzed/spicedb/pkg/errors"
+	"github.com/authzed/spicedb/pkg/schemadsl/compiler"
+)
+
+func rewriteError(ctx context.Context, err error) error {
```

Comment:
If everything implements the proper interface for producing metadata, I wonder if it's better to be explicit at the call-sites as to how we're converting. That way we can distinguish when two different calls can return the same error, but have different desired responses.

Reply:
`rewriteError` is per package, so it shouldn't need to be differentiated (yet anyway)

Reply:
The only issue I see with this approach is it'll cause developers to just thoughtlessly do `rewriteError(ctx, err)` and not think about what actually needs to happen.